wal/wal.go

```diff
@@ -429,7 +429,6 @@ func (w *WAL) flushPage(clear bool) error {
 	// No more data will fit into the page. Enqueue and clear it.
 	if clear {
 		p.alloc = pageSize // Write till end of page.
-		w.pageCompletions.Inc()
 	}
 	n, err := w.segment.Write(p.buf[p.flushed:p.alloc])
 	if err != nil {
```

Comment:
Line 437: What if the clear is true, and we do `Write` and `n` is not actually `p.alloc`? I think we would then drop some bytes, right? Now is the question if the implementation of `segment.Write` does partial writes in any case (`n`).  Worth to check?

Reply:
I know it's not your bug but: https://www.oreilly.com/library/view/97-things-every/9780596809515/ch08.html

Reply:
> I think we would then drop some bytes, right? 

Not sure what you mean by this. Can you add more details.

```
n, err := w.segment.Write(p.buf[p.flushed:p.alloc])
	if err != nil {
		return err
	}
	p.flushed += n
```
When `Write` returns an error tsdb will exit(leaving the WAL in corrupted state) and will run a repair  on the next start up.

Reply:
ping @bwplotka